src/mapping/GinkgoRadialBasisFctSolver.hpp

```diff
@@ -0,0 +1,533 @@
+    dRhs->sub_scaled(gko::lend(this->_scalarOne), gko::lend(subPolynomialContribution));
+
+    matrixQ_T->clear();
+    matrixQTQ->clear();
+    polynomialRHS->clear();
+    subPolynomialContribution->clear();
+  }
+
+  this->_solveRBFSystem(dRhs);
+
+  dRhs->clear();
+
+  auto output = gko::share(GinkgoVector::create(this->_deviceExecutor, gko::dim<2>{this->_matrixA->get_size()[0], this->_rbfCoefficients->get_size()[1]}));
+
+  this->_matrixA->apply(gko::lend(this->_rbfCoefficients), gko::lend(output));
+
+  if (polynomial == Polynomial::SEPARATE) {
+    auto addPolynomialContribution = gko::share(GinkgoVector::create(this->_deviceExecutor, gko::dim<2>{this->_matrixV->get_size()[0], 1}));
+    this->_matrixV->apply(gko::lend(this->_polynomialContribution), gko::lend(addPolynomialContribution));
+    output->add_scaled(gko::lend(this->_scalarOne), gko::lend(addPolynomialContribution));
+  }
+
+  this->_copyEvent.start();
+  output = gko::clone(this->_hostExecutor, output);
+
+  Eigen::VectorXd result(output->get_size()[0], 1);
+
+  for (Eigen::Index i = 0; i < result.rows(); ++i) {
+    result(i, 0) = output->at(i, 0);
+  }
+  this->_copyEvent.pause();
+
+  return result;
+}
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+Eigen::VectorXd GinkgoRadialBasisFctSolver<RADIAL_BASIS_FUNCTION_T>::solveConservative(const Eigen::VectorXd &rhsValues, Polynomial polynomial)
+{
+  return Eigen::VectorXd(1, 1);
+}
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+std::shared_ptr<gko::Executor> GinkgoRadialBasisFctSolver<RADIAL_BASIS_FUNCTION_T>::_hostExecutor = gko::ReferenceExecutor::create();
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+std::shared_ptr<gko::Executor> GinkgoRadialBasisFctSolver<RADIAL_BASIS_FUNCTION_T>::_ompExecutor = gko::OmpExecutor::create();
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+std::shared_ptr<gko::Executor> GinkgoRadialBasisFctSolver<RADIAL_BASIS_FUNCTION_T>::getReferenceExecutor() const
+{
+  return this->_hostExecutor;
+}
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+const std::shared_ptr<GinkgoMatrix> GinkgoRadialBasisFctSolver<RADIAL_BASIS_FUNCTION_T>::getEvaluationMatrix() const
+{
+  return this->_matrixA;
+}
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+void GinkgoRadialBasisFctSolver<RADIAL_BASIS_FUNCTION_T>::clear()
+{
+  if (nullptr == this->_rbfSystemMatrix) {
+    this->_rbfSystemMatrix = gko::share(GinkgoMatrix::create(this->_deviceExecutor, gko::dim<2>{0, 0}));
+    this->_matrixA         = gko::share(GinkgoMatrix::create(this->_deviceExecutor, gko::dim<2>{0, 0}));
+  } else {
+    this->_rbfSystemMatrix = gko::share(GinkgoMatrix::create(this->_deviceExecutor, gko::dim<2>{this->_rbfSystemMatrix->get_size()[0], this->_rbfSystemMatrix->get_size()[1]}));
+    this->_matrixA         = gko::share(GinkgoMatrix::create(this->_deviceExecutor, gko::dim<2>{this->_matrixA->get_size()[0], this->_matrixA->get_size()[1]}));
```

Comment:
Thanks for noticing, this was one of the very first functions available. We should replace it with calls to Ginkgo's `clear()` since it deallocates memory (I suspect memory leaks in the case here).